### PR TITLE
[CCFPCM-0433] enabled versioning (use makefile cmd to set version)

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[CCFPCM-0433](https://bcdevex.atlassian.net/browse/CCFPCM-0433)

Objective: 

- enable versioning
- update swagger module to make the requests to the proper endpoint
- remove global prefix and use enableVersioning 
- set version from package json
- include make commands to set version (can be used in a release flow/actions)
- use current version as default for all routes (this can be changed in the future if there is a need for version specific routing)

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -4,10 +4,12 @@ nodeLinker: node-modules
 
 plugins:
   - path: .yarn/plugins/@yarnpkg/plugin-workspace-tools.cjs
-    spec: '@yarnpkg/plugin-workspace-tools'
+    spec: "@yarnpkg/plugin-workspace-tools"
```
Comment (fw-noel) [inline .yarnrc.yml:7]: Why change these? Is this a linter thing?

**Review by fw-noel — COMMENTED** [review]
This looks pretty good, but a rundown in the README would be a good idea

File: .yarnrc.yml (hunk 1 above)
Comment (chelsea-EYDS) [reply .yarnrc.yml:7]: Yes - the pre-commit hook runs these so I don't see them in the diff prior to committing.

**Comment by chelsea-EYDS** [comment]
(set to DRAFT - changes pending)

File: apps/backend/src/app.config.ts [hunk 2]
```diff
@@ -96,8 +96,11 @@ export async function createNestApp(): Promise<{
   // Validation pipe
   app.useGlobalPipes(new TrimPipe(), new ValidationPipe(validationPipeConfig));
 
-  // Api prefix api/v1/
-  app.setGlobalPrefix(API_PREFIX);
+  app.enableVersioning({
+    type: VersioningType.URI,
+    prefix: 'api/v',
+    defaultVersion: process.env.APP_VERSION?.split('.')[0] ?? '1',
```
Comment (chelsea-EYDS) [inline apps/backend/src/app.config.ts:102]: We only want to update the api path in the case of a major version update

File: apps/backend/src/common/documentation.ts [hunk 3]
```diff
@@ -8,8 +8,8 @@ import { TransactionModule } from '../transaction/transaction.module';
 export const Documentation = (app: INestApplication) => {
   const options = new DocumentBuilder()
     .setTitle('PayCoCo API Docs')
+    .setVersion(process.env.APP_VERSION || '1.0.0')
```
Comment (chelsea-EYDS) [inline apps/backend/src/common/documentation.ts:11]: Here we can show the actual version, but the path will be v2, v1, etc

File: package.json [hunk 4]
```diff
@@ -9,6 +9,9 @@
     "node": "^18.14.0"
   },
   "scripts": {
+    "version:major": "yarn workspace @payment/backend version:major",
```
Comment (chelsea-EYDS) [inline package.json:12]: We can use these later as part of a release workflow? Running these cmds and then restarting the app will update the version displayed by swagger as well as the api paths (in the case of a major version update)

**Review by simensma-fresh — APPROVED** [review]
Looks great!

**Pull request merged**